tests/test_util.py

```diff
@@ -19,15 +21,17 @@ def test_generate_clks(self):
             self.assertEqual(len(clk[0]), 1024)
             self.assertEqual(clk[0].count(), clk[2])
 
-    def test_popcount_vector(self):
-        bas = [util.generate_bitarray(1024) for i in range(100)]
+# Generate bit arrays that are combinations of words 0, 1, 2^63, 2^64 - 1
+# of various lengths between 1 and 65 words.
+def test_popcount_vector():
+    for L in bitarray_utils.key_lengths:
+        yield check_popcount_vector, bitarray_utils.bitarrays_of_length(L)
```

Comment:
If this is the actual test shouldn't you be calling the `check_popcount_vector` function? Or am I missing where `test_popcount_vector` is called?

Reply:
Okay, this is now moot as I now use the `@pytest.mark.parametrize` decorator.

Reply:
great!